### PR TITLE
update contribution guide and build/test instructions

### PR DESCRIPTION
**Pull request by fiowro**

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| version-service | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/f958d3a7/version-service.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/f958d3a7/demand-backup.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/f958d3a7/scheduled-backup.log) |
| upgrade | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/f958d3a7/upgrade.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/f958d3a7/init-deploy.log) |
| smart-update | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/f958d3a7/smart-update.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/f958d3a7/scaling.log) |
| recreate | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/f958d3a7/recreate.log) |
| affinity | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/f958d3a7/affinity.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/f958d3a7/monitoring.log) |
| self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/f958d3a7/self-healing.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/f958d3a7/operator-self-healing.log) |
| clone-cluster | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/f958d3a7/clone-cluster.log) |
| tls-check | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/f958d3a7/tls-check.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/f958d3a7/users.log) |
| ns-mode | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/f958d3a7/ns-mode.log) |
| We run 16 out of 16|

commit: https://github.com/percona/percona-postgresql-operator/pull/289/commits/f958d3a77b5960471004d2ca87c7a41e3ada392a
image: `perconalab/percona-postgresql-operator:PR-289-f958d3a7-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-289-f958d3a7-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-289-f958d3a7-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-289-f958d3a7-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-289-f958d3a7-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-289-f958d3a7-pgo-deployer`

File: e2e-tests/README.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+Run the following commands to install the required components:
+
+```
+sudo yum -y install epel-release https://repo.percona.com/yum/percona-release-latest.noarch.rpm
+sudo yum -y install coreutils sed jq curl docker
+sudo curl -s -L https://github.com/mikefarah/yq/releases/download/3.4.1/yq_linux_amd64 -o /usr/bin/yq
+sudo chmod a+x /usr/bin/yq
+curl -s -L https://github.com/openshift/origin/releases/download/v3.11.0/openshift-origin-client-tools-v3.11.0-0cbc58b-linux-64bit.tar.gz \
+    | tar -C /usr/bin --strip-components 1 --wildcards -zxvpf - '*/oc' '*/kubectl'
```
Comment (cap1984) [inline e2e-tests/README.md:26]: no need in obsolete (3.11) tools. We are ok with 4.10

File: e2e-tests/README.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+You can also build the image and run your cluster in one command:
+
+```
+./e2e-tests/build-and-run
+```
+
+Running all tests at once can be done with the following command:
+
+```
+./e2e-tests/run
```
Comment (cap1984) [inline e2e-tests/README.md:82]: we have build script only in place. No similar scripts for test runs

File: e2e-tests/README.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+```
+
+(see how to configure the testing infrastructure [here](#using-environment-variables-to-customize-the-testing-process)).
+
+Tests can also be run one-by-one using the appropriate scripts (their names should be self-explanatory):
+
+```
+./e2e-tests/affinity/run
+./e2e-tests/clone-cluster/run
+./e2e-tests/data-migration-gcs/run
```
Comment (cap1984) [inline e2e-tests/README.md:92]: data-migration is an obsolete test. We are keeping it just for history. Here is an actual test list

`affinity
clone-cluster
demand-backup
init-deploy
monitoring
ns-mode
operator-self-healing
recreate
scaling
scheduled-backup
self-healing
smart-update
tls-check
upgrade
users
version-service`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/37827534/init-deploy.log) |
| version-service | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/37827534/version-service.log) |
| upgrade | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/37827534/upgrade.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/37827534/demand-backup.log) |
| smart-update | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/37827534/smart-update.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/37827534/scaling.log) |
| recreate | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/37827534/recreate.log) |
| affinity | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/37827534/affinity.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/37827534/monitoring.log) |
| self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/37827534/self-healing.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/37827534/operator-self-healing.log) |
| clone-cluster | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/37827534/clone-cluster.log) |
| tls-check | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/37827534/tls-check.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/37827534/users.log) |
| ns-mode | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/37827534/ns-mode.log) |
| We run 15 out of 16|

commit: https://github.com/percona/percona-postgresql-operator/pull/289/commits/378275345f1fce99d0f524d039fe525448dfd32f
image: `perconalab/percona-postgresql-operator:PR-289-37827534-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-289-37827534-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-289-37827534-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-289-37827534-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-289-37827534-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-289-37827534-pgo-deployer`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/11aa4b92/init-deploy.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/11aa4b92/scheduled-backup.log) |
| version-service | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/11aa4b92/version-service.log) |
| upgrade | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/11aa4b92/upgrade.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/11aa4b92/demand-backup.log) |
| smart-update | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/11aa4b92/smart-update.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/11aa4b92/scaling.log) |
| recreate | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/11aa4b92/recreate.log) |
| affinity | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/11aa4b92/affinity.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/11aa4b92/monitoring.log) |
| self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/11aa4b92/self-healing.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/11aa4b92/operator-self-healing.log) |
| clone-cluster | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/11aa4b92/clone-cluster.log) |
| tls-check | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/11aa4b92/tls-check.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/11aa4b92/users.log) |
| ns-mode | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/11aa4b92/ns-mode.log) |
| We run 16 out of 16|

commit: https://github.com/percona/percona-postgresql-operator/pull/289/commits/11aa4b929f32e7fb676a5f71a7768740c712f176
image: `perconalab/percona-postgresql-operator:PR-289-11aa4b92-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-289-11aa4b92-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-289-11aa4b92-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-289-11aa4b92-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-289-11aa4b92-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-289-11aa4b92-pgo-deployer`

**Review by cap1984 — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| upgrade | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/c8ac86b4/upgrade.log) |
| version-service | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/c8ac86b4/version-service.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/c8ac86b4/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/c8ac86b4/init-deploy.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/c8ac86b4/scheduled-backup.log) |
| smart-update | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/c8ac86b4/smart-update.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/c8ac86b4/scaling.log) |
| recreate | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/c8ac86b4/recreate.log) |
| affinity | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/c8ac86b4/affinity.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/c8ac86b4/monitoring.log) |
| self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/c8ac86b4/self-healing.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/c8ac86b4/operator-self-healing.log) |
| clone-cluster | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/c8ac86b4/clone-cluster.log) |
| tls-check | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/c8ac86b4/tls-check.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/c8ac86b4/users.log) |
| ns-mode | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-289/c8ac86b4/ns-mode.log) |
| We run 16 out of 16|

commit: https://github.com/percona/percona-postgresql-operator/pull/289/commits/c8ac86b46be55e7c0274e9f22131a1420c8edfd1
image: `perconalab/percona-postgresql-operator:PR-289-c8ac86b4-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-289-c8ac86b4-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-289-c8ac86b4-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-289-c8ac86b4-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-289-c8ac86b4-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-289-c8ac86b4-pgo-deployer`

**Review by spron-in — APPROVED** [review]

**Pull request merged**